src/-private/create-string-helper.ts

```diff
@@ -2,9 +2,9 @@ import { isHTMLSafe } from '@ember/template';
 
 import type { SafeString } from '@ember/template';
 
-export default function (
-  stringFunction: (input: string) => string | SafeString,
-): (args: [string | SafeString | undefined | null]) => string | SafeString {
+export default function <T = string>(
```

Comment:
```suggestion
export default function <T extends string | SafeString>(
```

Should this be something like this? Like we allow passing a string or SafeString in, right?